File: Cpp/1-Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        map<int, int> m1;
        int cnt = nums.size();
        for(int i= 0; i< cnt;i++){
            if(m1.count(nums[i]) == 1){
                ans.push_back(m1[nums[i]]);
                ans.push_back(i);
                return ans;
            }
            m1[target - nums[i]] = i;

        }
        return ans;
    }
};

File: python/1_Two-Sum.py
from typing import *
nums = [2,7,11,15]
target = 9


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = dict()
        for i in range(len(nums)):
            x = nums[i]
            if target-x not in hashtable:
                hashtable[x] = i
            else:
                return [hashtable[target-x],i]

if __name__ == '__main__':
    res = Solution()
    print(res.twoSum(nums,target))